x/auth/client/testutil/suite.go

```diff
@@ -1459,6 +1461,122 @@ func (s *IntegrationTestSuite) TestSignWithMultiSignersAminoJSON() {
 	require.Equal(sdk.NewCoins(val0Coin, val1Coin), queryRes.Balances)
 }
 
+func (s *IntegrationTestSuite) TestTipsWithFee() {
+	require := s.Require()
+	val := s.network.Validators[0]
+	val0Coin := sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(10))
+
+	testCases := []struct {
+		name      string
+		args      []string
+		expectErr bool
+	}{
+		{
+			"error with SIGN_MDOE_DIRECT_AUX and generate-only unset",
+			[]string{
+				fmt.Sprintf("--%s=%s", flags.FlagSignMode, flags.SignModeDirectAux),
+			},
+			true,
+		},
+		{
+			"no error with SIGN_MDOE_DIRECT_AUX mode and generate-only set",
+			[]string{
+				fmt.Sprintf("--%s=%s", flags.FlagSignMode, flags.SignModeDirectAux),
+				fmt.Sprintf("--%s=true", flags.FlagGenerateOnly),
+			},
+			false,
+		},
+		{
+			"no error with SIGN_MDOE_DIRECT_AUX mode and generate-only, tip flag set",
+			[]string{
+				fmt.Sprintf("--%s=%s", flags.FlagSignMode, flags.SignModeDirectAux),
+				fmt.Sprintf("--%s=true", flags.FlagGenerateOnly),
+				fmt.Sprintf("--%s=%s", flags.FlagTip, val0Coin.String()),
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			res, err := govtestutil.MsgSubmitProposal(
+				val.ClientCtx,
+				val.Address.String(),
+				"test",
+				"test desc",
+				govtypes.ProposalTypeText,
+				tc.args...,
+			)
+			if tc.expectErr {
+				require.Error(err)
+			} else {
+				require.NoError(err)
+				s.Require().False(strings.Contains(string(res.Bytes()), "\"tip\":null"))
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestTipsToFee() {
+	require := s.Require()
+	val := s.network.Validators[0]
+
+	kb := s.network.Validators[0].ClientCtx.Keyring
+	acc, _, err := kb.NewMnemonic("tipperAccount", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	require.NoError(err)
+
+	tipper, err := acc.GetAddress()
+	require.NoError(err)
+	tipperInitialBal := sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(10000))
+
+	feePayer := val.Address
+	fee := sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1000))
+	tip := sdk.NewCoin(fmt.Sprintf("%stoken", val.Moniker), sdk.NewInt(1000))
+
+	_, err = s.createBankMsg(val, tipper, sdk.NewCoins(tipperInitialBal))
+	require.NoError(err)
+	s.Require().NoError(s.network.WaitForNextBlock())
+
+	bal := s.getBalances(val.ClientCtx, tipper, tip.Denom)
+	s.Require().True(bal.Equal(tipperInitialBal.Amount))
+
+	// generate tx with --aux mode
+	res, err := govtestutil.MsgSubmitProposal(
+		val.ClientCtx,
+		tipper.String(),
+		"test",
+		"test desc",
+		govtypes.ProposalTypeText,
+		fmt.Sprintf("--%s=%s", flags.FlagSignMode, flags.SignModeDirectAux),
+		fmt.Sprintf("--%s=%s", flags.FlagTip, tip.String()),
```

Comment:
Can we use table tests here?

Some tests I would like to see:
- tipper uses AMINO, fee payer AMINO
- tipper uses AMINO, fee payer DIRECT
- tipper uses DIRECT_AUX, fee payer AMINO
- empty tip